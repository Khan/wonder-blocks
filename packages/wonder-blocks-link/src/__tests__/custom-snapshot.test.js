// @flow
import React from "react";
import renderer from "react-test-renderer";

import LinkCore from "../components/link-core";
import Link from "../components/link";

const defaultHandlers = {
    onClick: () => void 0,
    onMouseEnter: () => void 0,
    onMouseLeave: () => void 0,
    onMouseDown: () => void 0,
    onMouseUp: () => void 0,
    onTouchStart: () => void 0,
    onTouchEnd: () => void 0,
    onTouchCancel: () => void 0,
    onKeyDown: () => void 0,
    onKeyUp: () => void 0,
    onFocus: () => void 0,
    onBlur: () => void 0,
    tabIndex: 0,
};

describe("Link", () => {
    test.each`
        tabIndex
        ${-1}
        ${0}
        ${1}
    `("<Link tabIndex={$tabIndex}>", ({tabIndex}) => {
        const tree = renderer
            .create(
                <Link href="#" tabIndex={tabIndex}>
                    Click me
                </Link>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("LinkCore", () => {
    for (const kind of ["primary", "secondary"]) {
        for (const href of ["#", "#non-existent-link"]) {
            for (const light of kind === "primary" ? [true, false] : [false]) {
                for (const visitable of kind === "primary"
                    ? [true, false]
                    : [false]) {
                    for (const inline of [true, false]) {
                        for (const state of ["focused", "hovered", "pressed"]) {
                            const stateProps = {
                                focused: state === "focused",
                                hovered: state === "hovered",
                                pressed: state === "pressed",
                                waiting: false,
                            };
                            test(`kind:${kind} href:${href} light:${String(
                                light,
                            )} visitable:${String(visitable)} ${state}`, () => {
                                const tree = renderer
                                    .create(
                                        <LinkCore
                                            href="#"
                                            inline={inline}
                                            kind={kind}
                                            light={light}
                                            visitable={visitable}
                                            {...stateProps}
                                            {...defaultHandlers}
                                        >
                                            Click me
                                        </LinkCore>,
                                    )
                                    .toJSON();
                                expect(tree).toMatchSnapshot();
                            });
                        }
                    }
                }
            }
        }
    }
});
