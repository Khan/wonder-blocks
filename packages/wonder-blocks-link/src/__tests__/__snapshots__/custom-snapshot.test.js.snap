// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link <Link tabIndex={-1}> 1`] = `
<a
  className=""
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    {
      "color": "#1865f2",
      "cursor": "pointer",
      "outline": "none",
      "textDecoration": "none",
    }
  }
  tabIndex={-1}
>
  Click me
</a>
`;

exports[`Link <Link tabIndex={0}> 1`] = `
<a
  className=""
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    {
      "color": "#1865f2",
      "cursor": "pointer",
      "outline": "none",
      "textDecoration": "none",
    }
  }
  tabIndex={0}
>
  Click me
</a>
`;

exports[`Link <Link tabIndex={1}> 1`] = `
<a
  className=""
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    {
      "color": "#1865f2",
      "cursor": "pointer",
      "outline": "none",
      "textDecoration": "none",
    }
  }
  tabIndex={1}
>
  Click me
</a>
`;

exports[`LinkCore kind:primary href:# light:false visitable:false focused 1`] = `
<a
  className=""
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    {
      "color": "#1865f2",
      "cursor": "pointer",
      "outline": "none",
      "textDecoration": "underline currentcolor solid",
    }
  }
  tabIndex={0}
>
  Click me
</a>
`;

exports[`LinkCore kind:primary href:# light:false visitable:false hovered 1`] = `
<a
  className=""
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    {
      "color": "#1865f2",
      "cursor": "pointer",
      "outline": "none",
      "textDecoration": "underline currentcolor solid",
    }
  }
  tabIndex={0}
>
  Click me
</a>
`;

exports[`LinkCore kind:primary href:# light:false visitable:false pressed 1`] = `
<a
  className=""
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    {
      "color": "#1b50b3",
      "cursor": "pointer",
      "outline": "none",
      "textDecoration": "underline currentcolor solid",
    }
  }
  tabIndex={0}
>
  Click me
</a>
`;

exports[`LinkCore kind:primary href:# light:false visitable:true focused 1`] = `
<a
  className=""
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    {
      ":visited": {
        "color": "#8755ee",
      },
      "color": "#1865f2",
      "cursor": "pointer",
      "outline": "none",
      "textDecoration": "underline currentcolor solid",
    }
  }
  tabIndex={0}
>
  Click me
</a>
`;

exports[`LinkCore kind:primary href:# light:false visitable:true hovered 1`] = `
<a
  className=""
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    {
      ":visited": {
        "color": "#8755ee",
      },
      "color": "#1865f2",
      "cursor": "pointer",
      "outline": "none",
      "textDecoration": "underline currentcolor solid",
    }
  }
  tabIndex={0}
>
  Click me
</a>
`;

exports[`LinkCore kind:primary href:# light:false visitable:true pressed 1`] = `
<a
  className=""
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    {
      ":visited": {
        "color": "#6645b0",
      },
      "color": "#1b50b3",
      "cursor": "pointer",
      "outline": "none",
      "textDecoration": "underline currentcolor solid",
    }
  }
  tabIndex={0}
>
  Click me
</a>
`;

exports[`LinkCore kind:primary href:# light:true visitable:false focused 1`] = `
<a
  className=""
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    {
      "color": "#ffffff",
      "cursor": "pointer",
      "outline": "none",
      "textDecoration": "underline currentcolor solid",
    }
  }
  tabIndex={0}
>
  Click me
</a>
`;

exports[`LinkCore kind:primary href:# light:true visitable:false hovered 1`] = `
<a
  className=""
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    {
      "color": "#ffffff",
      "cursor": "pointer",
      "outline": "none",
      "textDecoration": "underline currentcolor solid",
    }
  }
  tabIndex={0}
>
  Click me
</a>
`;

exports[`LinkCore kind:primary href:# light:true visitable:false pressed 1`] = `
<a
  className=""
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    {
      "color": "#b5cefb",
      "cursor": "pointer",
      "outline": "none",
      "textDecoration": "underline currentcolor solid",
    }
  }
  tabIndex={0}
>
  Click me
</a>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:false visitable:false focused 1`] = `
<a
  className=""
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    {
      "color": "#1865f2",
      "cursor": "pointer",
      "outline": "none",
      "textDecoration": "underline currentcolor solid",
    }
  }
  tabIndex={0}
>
  Click me
</a>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:false visitable:false hovered 1`] = `
<a
  className=""
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    {
      "color": "#1865f2",
      "cursor": "pointer",
      "outline": "none",
      "textDecoration": "underline currentcolor solid",
    }
  }
  tabIndex={0}
>
  Click me
</a>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:false visitable:false pressed 1`] = `
<a
  className=""
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    {
      "color": "#1b50b3",
      "cursor": "pointer",
      "outline": "none",
      "textDecoration": "underline currentcolor solid",
    }
  }
  tabIndex={0}
>
  Click me
</a>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:false visitable:true focused 1`] = `
<a
  className=""
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    {
      ":visited": {
        "color": "#8755ee",
      },
      "color": "#1865f2",
      "cursor": "pointer",
      "outline": "none",
      "textDecoration": "underline currentcolor solid",
    }
  }
  tabIndex={0}
>
  Click me
</a>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:false visitable:true hovered 1`] = `
<a
  className=""
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    {
      ":visited": {
        "color": "#8755ee",
      },
      "color": "#1865f2",
      "cursor": "pointer",
      "outline": "none",
      "textDecoration": "underline currentcolor solid",
    }
  }
  tabIndex={0}
>
  Click me
</a>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:false visitable:true pressed 1`] = `
<a
  className=""
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    {
      ":visited": {
        "color": "#6645b0",
      },
      "color": "#1b50b3",
      "cursor": "pointer",
      "outline": "none",
      "textDecoration": "underline currentcolor solid",
    }
  }
  tabIndex={0}
>
  Click me
</a>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:true visitable:false focused 1`] = `
<a
  className=""
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    {
      "color": "#ffffff",
      "cursor": "pointer",
      "outline": "none",
      "textDecoration": "underline currentcolor solid",
    }
  }
  tabIndex={0}
>
  Click me
</a>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:true visitable:false hovered 1`] = `
<a
  className=""
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    {
      "color": "#ffffff",
      "cursor": "pointer",
      "outline": "none",
      "textDecoration": "underline currentcolor solid",
    }
  }
  tabIndex={0}
>
  Click me
</a>
`;

exports[`LinkCore kind:primary href:#non-existent-link light:true visitable:false pressed 1`] = `
<a
  className=""
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    {
      "color": "#b5cefb",
      "cursor": "pointer",
      "outline": "none",
      "textDecoration": "underline currentcolor solid",
    }
  }
  tabIndex={0}
>
  Click me
</a>
`;

exports[`LinkCore kind:secondary href:# light:false visitable:false focused 1`] = `
<a
  className=""
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    {
      "color": "#1865f2",
      "cursor": "pointer",
      "outline": "none",
      "textDecoration": "underline currentcolor solid",
    }
  }
  tabIndex={0}
>
  Click me
</a>
`;

exports[`LinkCore kind:secondary href:# light:false visitable:false hovered 1`] = `
<a
  className=""
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    {
      "color": "#1865f2",
      "cursor": "pointer",
      "outline": "none",
      "textDecoration": "underline currentcolor solid",
    }
  }
  tabIndex={0}
>
  Click me
</a>
`;

exports[`LinkCore kind:secondary href:# light:false visitable:false pressed 1`] = `
<a
  className=""
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    {
      "color": "#1b50b3",
      "cursor": "pointer",
      "outline": "none",
      "textDecoration": "underline currentcolor solid",
    }
  }
  tabIndex={0}
>
  Click me
</a>
`;

exports[`LinkCore kind:secondary href:#non-existent-link light:false visitable:false focused 1`] = `
<a
  className=""
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    {
      "color": "#1865f2",
      "cursor": "pointer",
      "outline": "none",
      "textDecoration": "underline currentcolor solid",
    }
  }
  tabIndex={0}
>
  Click me
</a>
`;

exports[`LinkCore kind:secondary href:#non-existent-link light:false visitable:false hovered 1`] = `
<a
  className=""
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    {
      "color": "#1865f2",
      "cursor": "pointer",
      "outline": "none",
      "textDecoration": "underline currentcolor solid",
    }
  }
  tabIndex={0}
>
  Click me
</a>
`;

exports[`LinkCore kind:secondary href:#non-existent-link light:false visitable:false pressed 1`] = `
<a
  className=""
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    {
      "color": "#1b50b3",
      "cursor": "pointer",
      "outline": "none",
      "textDecoration": "underline currentcolor solid",
    }
  }
  tabIndex={0}
>
  Click me
</a>
`;
